### PR TITLE
Add Mistral NeMo (fix)

### PR DESCRIPTION
**Pull request by Blaizzy**
This PR adds support for Mistral NeMo
Twitter: @Prince_Canuma

### CODE REVIEW

File: llms/mlx_lm/models/llama.py [hunk 1]
```diff
@@ -45,7 +46,8 @@ def __init__(self, args: ModelArgs):
         self.n_heads = n_heads = args.num_attention_heads
         self.n_kv_heads = n_kv_heads = args.num_key_value_heads
 
-        head_dim = args.hidden_size // n_heads
+        self.head_dim = head_dim = args.head_dim if args.head_dim else args.hidden_size // n_heads
```
Comment (awni) [inline llms/mlx_lm/models/llama.py:49]: Very minor nit, this is a nice idiom you can use in Python for when something is `None`

```suggestion
        self.head_dim = head_dim = args.head_dim or args.hidden_size // n_heads
```

Comment (Blaizzy) [reply llms/mlx_lm/models/llama.py]: Cool, I didn't know that.

**Comment by Blaizzy** [comment]
@awni  I'm getting a KV cache broadcast error: 

```
(mlx_code) prince_canuma@MacBook-Pro grpc % python -m mlx_lm.generate --model mlx-community/Mistral-Nemo-Instruct-2407-4bit --prompt "hi"
Fetching 9 files: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 9/9 [00:00<00:00, 26087.59it/s]
==========
Prompt: <s>[INST] hi[/INST]
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/mlx_code/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/homebrew/Caskroom/miniconda/base/envs/mlx_code/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/generate.py", line 161, in <module>
    main()
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/generate.py", line 148, in main
    generate(
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/utils.py", line 307, in generate
    for (token, logprobs), n in zip(
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/utils.py", line 218, in generate_step
    y, logprobs = _step(y)
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/utils.py", line 201, in _step
    logits = model(y[None], cache=cache)
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/models/llama.py", line 199, in __call__
    out = self.model(inputs, cache)
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/models/llama.py", line 180, in __call__
    h = layer(h, mask, cache=c)
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/models/llama.py", line 142, in __call__
    r = self.self_attn(self.input_layernorm(x), mask, cache)
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/models/llama.py", line 92, in __call__
    keys, values = cache.update_and_fetch(keys, values)
  File "/Users/prince_canuma/Documents/Projects/LLMs/mlx-lm/llms/mlx_lm/models/base.py", line 47, in update_and_fetch
    self.keys[..., prev : self.offset, :] = keys
ValueError: Shapes (1,8,6,128) and (1,8,6,160) cannot be broadcast.
```

**Comment by awni** [comment]
Update [this section](https://github.com/ml-explore/mlx-examples/blob/260ee55e8a44329211a2973d53ada77e46f90042/llms/mlx_lm/models/llama.py#L216-L218)

**Comment by Blaizzy** [comment]
Thanks!

**Comment by Blaizzy** [comment]
@awni It's done ✅

**Comment by Blaizzy** [comment]
@awni, there was a tiny mistake with a file that should have been ignored, but it's fixed :)

**Review by awni — APPROVED** [review]
Thanks for the addition!

**Pull request merged**